android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java

```diff
@@ -217,136 +236,111 @@ public void setConfiguration(BitmovinPlayerView view, ReadableMap config) throws
         this.doLayoutChanges(fullscreen);
     }
 
-    private void doRotation(boolean fullScreen)
-    {
-        int rotation = this._activity.getWindowManager().getDefaultDisplay().getRotation();
-
-        if (fullScreen)
-        {
-            // System.out.println("XXX: doRotation fullscreen.");
-            switch (rotation)
-            {
-                case Surface.ROTATION_270:
-                    this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE);
-                    break;
-
-                default:
-                    this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
-            }
-        }
-        else
-        {
-            // System.out.println("XXX: doRotation no fullscreen.");
-            this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
-        }
-    }
-
-    private void doSystemUiVisibility(final boolean fullScreen)
-    {
-        this._decorView.post(new Runnable()
-        {
-            @Override
-            public void run()
-            {
-                int uiParams = FullscreenUtil.getSystemUiVisibilityFlags(fullScreen, true);
-                _decorView.setSystemUiVisibility(uiParams);
-            }
+    //private void doRotation(boolean fullScreen) {
+    //    int rotation = this._activity.getWindowManager().getDefaultDisplay().getRotation();
+    //    if (fullScreen) {
+    //        switch (rotation) {
+    //            case Surface.ROTATION_270:
+    //                this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE);
+    //                break;
+    //
+    //            default:
+    //                this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+    //        }
+    //    } else {
+    //        this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+    //    }
+    //}
+
+    private void doSystemUiVisibility(final boolean fullScreen) {
+        _decorView.post(() -> {
+            int uiParams = FullscreenUtil.getSystemUiVisibilityFlags(fullScreen, true);
+            _decorView.setSystemUiVisibility(uiParams);
         });
     }
 
-    private void doLayoutChanges(final boolean fullscreen)
-    {
-        // System.out.println("XXX: doLayoutChanges.");
-        Looper mainLooper = Looper.getMainLooper();
-        boolean isMainLooperAlready = Looper.myLooper() == mainLooper;
-
-
-        UpdateLayoutRunnable updateLayoutRunnable = new UpdateLayoutRunnable(this._activity, fullscreen);
-
-        if (isMainLooperAlready)
-        {
-            updateLayoutRunnable.run();
-        }
-        else
-        {
-            Handler handler = new Handler(mainLooper);
-            handler.post(updateLayoutRunnable);
-        }
+    private void doLayoutChanges(final boolean fullscreen) {
+        _decorView.post(new UpdateLayoutRunnable(fullscreen));
```

Comment:
Hopefully this code works on all android devices.

Reply:
`doSystemUiVisibility` used the same method to run on UI thread and it's called at the same time so I think it should be OK.

Actually I tried not calling either function and fullscreen still worked for me